include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1523,14 +1523,14 @@ struct __parallel_find_forward_tag
 };
 
 // Tag for __parallel_find_or to find the last element that satisfies predicate
-template <typename _RangeType>
+template <typename... _Ranges>
 struct __parallel_find_backward_tag
```

Comment:
It seems there is at least one place where it should be used with multiple ranges, but it isn't. (`__pattern_find_end`).
Should we add it there?

Reply:
Now it is.
Dmitriy asked the same question.
Please see 
https://github.com/uxlfoundation/oneDPL/pull/2100#discussion_r2047375158
https://github.com/uxlfoundation/oneDPL/pull/2100#discussion_r2047327289

Reply:
I saw those conversations but understood them to be about `__parallel_find_forward_tag` and `single_match_pred_by_idx` respectively.  
`__parallel_find_backward_tag` is only used in 3 places in the code, twice in `__parallel_find` overloads, and once in `__pattern_find_end`.  All three of those locations are unchanged and provide a single template argument.  

I mean it is OK if we are just adding infrastructure, but my question is more pointed:
Should we add both ranges within `__pattern_find_end` as arguments to `__parallel_find_backend_tag`?